code42_on_poll_connector.py

```diff
@@ -0,0 +1,287 @@
+import dateutil.parser
+import phantom.app as phantom
+from py42.sdk.queries.fileevents.file_event_query import FileEventQuery
+from py42.sdk.queries.fileevents.filters import (
+    ExposureType,
+    DeviceUsername,
+    Actor,
+    EventTimestamp,
+    EventType,
+    FileCategory,
+)
+
+from code42_util import get_thirty_days_ago, build_alerts_query
+
+"""The contents of this module that related to mapping alert observations to file events borrows heavily from the
+Code42 Cortex XSOAR integration.
+"""
+
+
+def get_file_category_value(key):
+    # Meant to handle all possible cases
+    key = key.lower().replace("-", "").replace("_", "")
+    category_map = {
+        "sourcecode": FileCategory.SOURCE_CODE,
+        "audio": FileCategory.AUDIO,
+        "executable": FileCategory.EXECUTABLE,
+        "document": FileCategory.DOCUMENT,
+        "image": FileCategory.IMAGE,
+        "pdf": FileCategory.PDF,
+        "presentation": FileCategory.PRESENTATION,
+        "script": FileCategory.SCRIPT,
+        "spreadsheet": FileCategory.SPREADSHEET,
+        "video": FileCategory.VIDEO,
+        "virtualdiskimage": FileCategory.VIRTUAL_DISK_IMAGE,
+        "archive": FileCategory.ZIP,
+    }
+    return category_map.get(key, "UNCATEGORIZED")
+
+
+class Code42OnPollConnector:
+    def __init__(self, connector, client, state):
+        super(Code42OnPollConnector, self).__init__()
+        self._connector = connector
+        self._client = client
+        self._state = state or {}
+
+    def handle_on_poll(self, param, action_result):
+        last_time = self._state.get("last_time")
```

Comment:
Is there any Phantom documentation you found for how the polling actions work, or did you just work from the examples of other apps? I'm a bit confused as to how `last_time` gets set on the state object, since you aren't setting it here in the action handler.

Reply:
I have not worked out `last_time` yet. And there docs for this feature are very much lacking, I am trying to reverse engineer the Jira `on_poll` which does a similar thing